moveit_planners/pilz_industrial_motion_planner/src/trajectory_functions.cpp

```diff
@@ -198,13 +198,17 @@ bool pilz_industrial_motion_planner::generateJointTrajectory(
   ros::Time generation_begin = ros::Time::now();
 
   // generate the time samples
-  const double epsilon = 10e-06;  // avoid adding the last time sample twice
-  std::vector<double> time_samples;
-  for (double t_sample = 0.0; t_sample < trajectory.Duration() - epsilon; t_sample += sampling_time)
+  std::vector<double> time_samples = { 0.0 };
+  double actual_sampling_time = sampling_time;
+  if (trajectory.Duration() > 10e-06)
```

Comment:
Better reintroduce a `const` variable for this threshold, e.g. `epsilon` as before.